Nightwatch/Program.fs

```diff
@@ -71,19 +74,20 @@ let private configureResourceRegistry() =
     Registry.create resourceFactories
 
 let private readConfiguration path factories : Result<Resource seq, InvalidConfiguration seq> =
-    async {
+    let readTask = task {
         let! resources = Configuration.read factories FileSystem.system path
         let (results, errors) = splitResults resources
         return
             if Seq.isEmpty errors
             then Ok results
             else Error errors
-    } |> Async.RunSynchronously
+    }
+    readTask.Result
```

Comment:
I suggest to return `Task` from this method and let the caller (i.e. the `main` function) to decide if it wants to get the `Result` synchronously.